testng-collections/src/main/java/org/testng/util/Strings.java

```diff
@@ -60,8 +38,8 @@ public static String valueOf(Map<?, ?> m) {
     return m.values().stream().map(Object::toString).collect(Collectors.joining(" "));
   }
 
-  /** @deprecated - This is deprecated of TestNG <code>7.6.0</code> */
-  @Deprecated
+  // Don't remove this method. This is being called as part of the Gradle build. Removing this
```

Comment:
Why not updating the gradle build?

Reply:
Could you dig a bit more please? Do you have the exact line in the gradle codebase that is using this method?

Reply:
@juherr - Here's the complete stack trace

```
org.gradle.api.internal.tasks.testing.TestSuiteExecutionException: Could not complete execution for Gradle Test Executor 1.
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:63)
	at java.base@11.0.11/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.11/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.11/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.11/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.IllegalAccessError: class org.testng.internal.annotations.IgnoreListener tried to access private method 'java.lang.String org.testng.util.Strings.join(java.lang.String, java.lang.String[])' (org.testng.internal.annotations.IgnoreListener and org.testng.util.Strings are in unnamed module of loader 'app')
	at org.testng.internal.annotations.IgnoreListener.findAnnotation(IgnoreListener.java:80)
	at org.testng.internal.annotations.IgnoreListener.ignoreTestAtClass(IgnoreListener.java:44)
	at org.testng.internal.annotations.IgnoreListener.transform(IgnoreListener.java:35)
	at org.testng.internal.annotations.DefaultAnnotationTransformer.transform(DefaultAnnotationTransformer.java:20)
	at org.testng.internal.annotations.JDK15AnnotationFinder.transform(JDK15AnnotationFinder.java:165)
	at org.testng.internal.annotations.JDK15AnnotationFinder.lambda$findAnnotation$0(JDK15AnnotationFinder.java:236)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at org.testng.internal.annotations.JDK15AnnotationFinder.findAnnotation(JDK15AnnotationFinder.java:234)
	at org.testng.internal.annotations.JDK15AnnotationFinder.findAnnotation(JDK15AnnotationFinder.java:109)
	at org.testng.internal.TestNGClassFinder.isTestNGClass(TestNGClassFinder.java:280)
	at org.testng.internal.TestNGClassFinder.isNotTestNGClass(TestNGClassFinder.java:265)
	at org.testng.internal.TestNGClassFinder.processClass(TestNGClassFinder.java:98)
	at org.testng.internal.TestNGClassFinder.<init>(TestNGClassFinder.java:71)
	at org.testng.TestRunner.initMethods(TestRunner.java:437)
	at org.testng.TestRunner.init(TestRunner.java:336)
	at org.testng.TestRunner.init(TestRunner.java:289)
	at org.testng.TestRunner.<init>(TestRunner.java:180)
	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:613)
	at org.testng.SuiteRunner.init(SuiteRunner.java:178)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:112)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1306)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1282)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1131)
	at org.testng.TestNG.runSuites(TestNG.java:1069)
	at org.testng.TestNG.run(TestNG.java:1037)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.runTests(TestNGTestClassProcessor.java:141)
	at org.gradle.api.internal.tasks.testing.testng.TestNGTestClassProcessor.stop(TestNGTestClassProcessor.java:90)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	... 18 more
```

Reply:
I think Gradle is using an older version of TestNG embedded into it wherein the IgnoreListener (below snippet) calls this removed method

```java
  private static Ignore findAnnotation(Package testPackage) {
    if (testPackage == null) {
      return null;
    }
    Ignore result = testPackage.getAnnotation(Ignore.class);
    if (result != null) {
      return result;
    }
    String[] parts = testPackage.getName().split("\\.");
    String[] parentParts = Arrays.copyOf(parts, parts.length - 1);
    String parentPackageName = String.join(".", parentParts);
    if (parentPackageName.isEmpty()) {
      return null;
    }
    return findAnnotation(Package.getPackage(parentPackageName));
  }
```

Reply:
According to the traces, the issue comes from `org.testng.internal.annotations.IgnoreListener.findAnnotation(IgnoreListener.java:80)` from the app module.
But the code doesn't use strings 🙃
Maybe a broken cache somewhere?

Reply:
@juherr - No that's not the case here.
I deleted the gradle cache at `~/.gradle/caches` and ran the tests again. The result is the same, which kind of gives me the feel that there's an embedded version of TestNG being used somewhere. But the error is consistent and so I don't see any other alternative but to leave that method there until gradle uses the newer version of TestNG maybe ?

Reply:
@juherr - I agree with you that at-least theoretically speaking Gradle would work just like the surefire plugin, wherein it would have an adapter which would call in the implementation found in the classpath, but something is missing here. 

But that should not stop us from proceeding further with this PR. Maybe we can merge this PR and you could help me out with just this one case later ? I just don't want to leave PRs around and would like to close out on PRs before moving to the next one.

Reply:
What is the emergency?

But right, you can open an issue if you prefer.

Reply:
@juherr - 

> What is the emergency?

It's not the emergency. I don't want PR's to go stale because we found side effect problems due to the code change.

> you can open an issue if you prefer.

I have raised an issue that explicitly tracks this discrepancy. https://github.com/cbeust/testng/issues/2766